web/api/v1/api.go

```diff
@@ -805,6 +805,71 @@ func rulesAlertsToAPIAlerts(rulesAlerts []*rules.Alert) []*Alert {
 	return apiAlerts
 }
 
+type metadataSet map[metadata]struct{}
+
+func (ms metadataSet) MarshalJSON() ([]byte, error) {
+	json := jsoniter.ConfigCompatibleWithStandardLibrary
+	m := []metadata{}
+
+	for s := range ms {
+		m = append(m, s)
+	}
+
+	return json.Marshal(m)
+}
+
+type metadata struct {
+	Type textparse.MetricType `json:"type"`
+	Help string               `json:"help"`
+	Unit string               `json:"unit"`
+}
+
+func (api *API) metricMetadata(r *http.Request) apiFuncResult {
+	metrics := map[string]metadataSet{}
+
+	limit := -1
+	if s := r.FormValue("limit"); s != "" {
+		var err error
+		if limit, err = strconv.Atoi(s); err != nil {
+			return apiFuncResult{nil, &apiError{errorBadData, errors.New("limit must be a number")}, nil, nil}
+		}
+	}
+
+	for _, tt := range api.targetRetriever.TargetsActive() {
+		for _, t := range tt {
+			for _, mm := range t.MetadataList() {
+				m := metadata{Type: mm.Type, Help: mm.Help, Unit: mm.Unit}
+				ms, ok := metrics[mm.Metric]
```

Comment:
It's odd to have the if for this many lines down, and the comment isn't really needed as this is a standard pattern.

Reply:
Done.